tests/lua/lua-scflowvarset/setflowvar.lua

```diff
@@ -0,0 +1,23 @@
+local flowvarlib = require("suricata.flowvar")
+
+function init()
+   local flowvar = flowvarlib.register("test_var")
+   return {}
+end
+
+function thread_init()
+   flowvar = flowvarlib.get("test_var")
+end
```

Comment:
A good example of what I mention here: https://github.com/OISF/suricata/pull/13093/commits/647cee7ade29bcedb5c6e036346ea736138093ff

If I register the keyword in `thread_init`, the rule matching all works as expected, however the flowvar is not logged in the metadata. That only happens when its registered earlier than `thread_init`, for example in `init`.  So that is why we have this process of:
1) register first
2) get handle to flowvar in each thread